common/lib/xmodule/xmodule/course_metadata_utils.py

```diff
@@ -105,12 +106,17 @@ def may_certify_for_course(certificates_display_behavior, certificates_show_befo
         certificates_show_before_end (bool): whether user can download the
             course's certificates before the course has ended.
         has_ended (bool): Whether the course has ended.
+        certificate_available_date (datetime): the date the certificate is available on for the course.
     """
     show_early = (
         certificates_display_behavior in ('early_with_info', 'early_no_info')
         or certificates_show_before_end
     )
-    return show_early or has_ended
+    past_availability_date = (
+        certificate_available_date
+        and certificate_available_date < datetime.now(utc)
+    )
+    return past_availability_date or show_early or has_ended
```

Comment:
Did you clarify this logic with product? I know that `show_early` has precedence over the cert available date, but I didn't think that we were showing certs upon course end unless the cert available date had passed as well. In fact, I'm not sure what the role of `has_ended` would be in the method at all now that `certificate_available_date` is defined.